templates/default/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+//! Note: This contract is not meant for production, it is used to illustrate
+//! how to use the standard library and the tooling Concordium provides. There
+//! is no claim that the logic of the contract is reasonable, or safe.
+//!
+//! # A Concordium V1 smart contract
+use concordium_std::*;
+use core::fmt::Debug;
+
+/// Your smart contract state.
+#[derive(Serialize, SchemaType, Clone)]
+pub struct State {
+    // Your state
+}
+
+/// Your smart contract errors.
+#[derive(Debug, PartialEq, Eq, Reject, Serial, SchemaType)]
+enum Error {
+    /// Failed parsing the parameter.
+    #[from(ParseError)]
+    ParseParamsError,
+    /// Your error
+    YourError,
+}
+
+/// Init function that creates a new smart contract.
+#[init(contract = "{{crate_name}}")]
+fn init<S: HasStateApi>(
+    _ctx: &impl HasInitContext,
+    _state_builder: &mut StateBuilder<S>,
+) -> InitResult<State> {
+    // Your code
+
+    Ok(State {})
+}
+
+/// Receive function. The input parameter is the boolean variable `throw_error`.
+///  If `throw_error == true`, the receive function will throw a custom error.
+///  If `throw_error == false`, the receive function executes successfully.
+#[receive(
+    contract = "{{crate_name}}",
+    name = "receive",
+    parameter = "bool",
+    error = "Error",
+    mutable
+)]
+fn receive<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State, StateApiType = S>,
+) -> Result<(), Error> {
+    // Your code
+
+    let throw_error = ctx.parameter_cursor().get()?; // Returns Error::ParseError on failure
+    if throw_error {
+        Err(Error::YourError)
+    } else {
+        Ok(())
+    }
+}
+
+/// View function that returns the content of the state.
+#[receive(contract = "{{crate_name}}", name = "view", return_value = "State")]
+fn view<'a, 'b, S: HasStateApi>(
+    _ctx: &'a impl HasReceiveContext,
+    host: &'b impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<&'b State> {
+    Ok(host.state())
+}
+
+#[concordium_cfg_test]
+mod tests {
+    use super::*;
+    use test_infrastructure::*;
+
+    type ContractResult<A> = Result<A, Error>;
+
+    fn expect_error<E, T>(expr: Result<T, E>, err: E, msg: &str)
+    where
+        E: Eq + Debug,
+        T: Debug,
+    {
+        let actual = expr.expect_err_report(msg);
+        claim_eq!(actual, err);
+    }
+
+    #[concordium_test]
+    /// Init test
```

Comment:
Could we do a bit better here
```suggestion
    /// Test that initializing the contract succeeds with some state.
```